File: Sem4/MPP_Templates/ZboruriCSharp/services/IFlightServer.cs
using System;
using System.Collections.Generic;
using model;

namespace services
{
    public interface IFlightServer
    {
        void LogIn(Angajat angajat, IFlightObserver observer);

        void LogOut(Angajat angajat, IFlightObserver observer);

        Client FindClient(String numeClient, String adresaClient);

        Zbor FindZbor(String destinatie, String data, String ora);

        int FindIndex();

        void CumparaBilet(Bilet bilet, int idZbor, int locuri);

        IEnumerable<Zbor> GetZboruri();

        IEnumerable<ZborDTO> GetZboruriCautare(String destinatie, String plecare);
    }
}

File: Sem4/MPP_Templates/Zboruri/FlightServices/src/main/java/services/IFlightServer.java
package services;

import flight.domain.*;
import flight.domain.*;

public interface IFlightServer {
    void logIn(Angajat angajat, IFlightObserver observer) throws FlightException;

    void logOut(Angajat angajat, IFlightObserver observer) throws FlightException;

    Client findClient(String numeClient, String adresaClient) throws FlightException;

    Zbor findZbor(String destinatie, String data, String ora) throws FlightException;

    Integer findIndex() throws FlightException;

    void cumparaBilet(Bilet bilet, int idZbor) throws FlightException;

    Iterable<Zbor> getZboruri() throws FlightException;

    Iterable<ZborDTO> getZboruriCautare(String destinatie, String plecare) throws FlightException;

    void zborUpdate(Integer zborID, Integer locuri) throws FlightException;
}
